src/utils.ts

```diff
@@ -41,4 +41,28 @@ export class Utils {
     const cidVersion = 1
     return CID.create(cidVersion, ETH_TX_CODE, multihash)
   }
+
+  /**
+   * Coalesce a possible null value to a value or error
+   * @param func
+   * @param error
+   */
+  static coalesce<T>(func: () => T | null | undefined, error: string): T {
```

Comment:
`checkNotNull`?  it wasn't super obvious to me from reading the calling code what "coalesce" meant